### PR TITLE
Fix the bug where File > Quit on the WebXDC window closes DC

### LINKED ISSUE #3988: File » Quit from an app quits all of DeltaChat
- OS: Debian stable (12.5) amd64
- Delta Chat Version: 1.46.1 (installed via .deb)

If i open a webxdc app, it opens as a new window.

If, from that window, i choose "File » Quit", then all of DeltaChat closes, not just the webxdc app.  (closing the webxdc window via the window manager/compositor just closes the webxdc app, not all of DeltaChat

File » Quit in the webxdc app should just close the specific app, not the whole DeltaChat.

### PR DESCRIPTION
**Pull request by farooqkz**
This PR fixes #3988 

It also implements [the suggestion](https://github.com/deltachat/deltachat-desktop/issues/3988#issuecomment-2198590315) in the same thread.

### CODE REVIEW

**Comment by Simon-Laux** [comment]
> a i do not know much about react.

this is just electron code, nothing related to react.

File: src/main/windows/main.ts [hunk 1]
```diff
@@ -51,37 +51,31 @@ export function init(options: { hidden: boolean }) {
...
-    })
-  ))
-  main_window.filePathWhiteList = []
-
-  initMinWinDimensionHandling(
-    main_window,
-    defaults.minWidth,
-    defaults.minHeight
-  )
+  const _window = (window = <ExtendedBrowserWindow>new electron.BrowserWindow({
```
Comment (Simon-Laux) [inline src/main/windows/main.ts:54]: why not rename it to `mainWindow`? `_` makes me wonder wether it's unused or private?

**Comment by Simon-Laux** [comment]
> a general note: due to added whitespace, renamings and reformattings, i assume it is also harder than needed to see the relevant changes and to review.

yes made it harder to review, please try to separate that in the future, if the pr diff is small it's easier to review.

File: src/main/menu.ts [hunk 2]
```diff
@@ -179,20 +179,28 @@ export function getFileMenu(
...
     label: tx('global_menu_file_desktop'),
-    submenu: [
-      {
-        label: tx('menu_settings'),
-        click: () => {
-          mainWindow.send('showSettingsDialog')
+    submenu: (() => {
+      let result = [
+        {
+          label: tx('global_menu_file_quit_desktop'),
```
Comment (Simon-Laux) [inline src/main/menu.ts:185]: I think this should be called "close window" and also I'm not sure if Ctrl+q is appropriate for it. 
I just have some doubts, I don't know the answer for linux and windows
CC @r10s @WofWca

Comment (nicodh) [reply src/main/menu.ts:185]: On Linux Ctr+q is correct. Windows I'm also not sure

**Review by nicodh — APPROVED** [review]
Reviewed by testing (on Linux)

File: src/main/menu.ts (hunk 2 above)
Comment (r10s) [reply src/main/menu.ts:185]: for windows, ctrl-q to exit is no mistake. and alt-f4, of couse

Comment (WofWca) [reply src/main/menu.ts:185]: It's fine I think (as a Windows user)

Comment (Simon-Laux) [reply src/main/menu.ts:185]: so then on windows and linux both `ctrl+q` and `ctrl+w` close the window, while on macOS `cmd+w` closes just window and `cmd+q` closes the whole app?

Comment (farooqkz) [reply src/main/menu.ts:185]: So I guess we need a new translation string for "close window"?

**Comment by farooqkz** [comment]
> thanks a lot for taking care!
> 
> i tested the pr on mac with the global menu bar to make sure, "Quit" from the "Application" menu still quits - looks fine. i cannot test on linux/windows.
> 
> i also cannot say much to the code, as i do not know much about react. a general note: due to added whitespace, renamings and reformattings, i assume it is also harder than needed to see the relevant changes and to review. it is better to put unrelated changes to dedicated PRs (or, for small things, at least to dedicated commits and mentioning that in the PR description)

Did you just say the App menu quits the whole DeltaChat app on Mac?

**Comment by farooqkz** [comment]
> > a general note: due to added whitespace, renamings and reformattings, i assume it is also harder than needed to see the relevant changes and to review.
> 
> yes made it harder to review, please try to separate that in the future, if the pr diff is small it's easier to review.

It isn't my fault. It's my editor's bad.

**Comment by farooqkz** [comment]
@r10s Can you please test the latest commit on Mac and see if the issue persists?

**Comment by farooqkz** [comment]
@r10s Can you please test the latest commit on Mac and see if the issue persists?

File: src/main/menu.ts (hunk 2 above)
Comment (Simon-Laux) [reply src/main/menu.ts:186]: > So I guess we need a new translation string for "close window"?

no, already exists in `_locales/en.xml` as `close_window`

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,18 +2,25 @@
 
 ## [Unreleased][unreleased]
 
+## Added
+- More shortcuts to switch between chats: `Ctrl + PageDown`, `Ctrl + PageUp`, `Ctrl + Tab`, `Ctrl + Shift + Tab` #3984
```
Comment (WofWca) [inline CHANGELOG.md:6]: Still something wrong with the CHANGELOG or am I wrong.

File: src/main/menu.ts [hunk 4]
```diff
@@ -206,7 +218,7 @@ export function getFileMenu(
             refresh()
           }
         },
-        accelerator: isMac ? 'Cmd+w' : 'Ctrl+w',
```
Comment (WofWca) [inline src/main/menu.ts:209]: Ugh, unrelated changes again... But fine, it's not too big.

**Comment by nicodh** [comment]
@farooqkz there is no menu on webxdc widgets on Mac. That's why @r10s tested only the global menu. No need to wait for him to test anything here on Mac.

File: src/main/menu.ts [hunk 5]
```diff
@@ -166,7 +167,7 @@ export function getAppMenu(
         : []),
       {
         label: tx('global_menu_file_quit_desktop'),
-        role: 'quit',
+        click: () => window?.close(),
         accelerator: 'Cmd+q',
       },
     ],
```
Comment (Simon-Laux) [inline src/main/menu.ts:173]: wrong, this menu is only for mac, so please revert this change

File: src/main/menu.ts [hunk 6]
```diff
@@ -166,7 +167,7 @@ export function getAppMenu(
         : []),
       {
         label: tx('global_menu_file_quit_desktop'),
-        role: 'quit',
+        click: () => window?.close(),
```
Comment (Simon-Laux) [inline src/main/menu.ts:170]:
```suggestion
        role: 'quit',
```

**Pull request merged**